session/pingpong/hermes_channel_repository.go

```diff
@@ -252,6 +274,74 @@ func (hcr *HermesChannelRepository) handleNodeStart(payload nodeEvent.Payload) {
 	hcr.fetchKnownChannels(config.GetInt64(config.FlagChainID))
 }
 
+func (hcr *HermesChannelRepository) handleIdentityUnlock(payload identity.AppEventIdentityUnlock) {
+	hermes, err := hcr.addressProvider.GetActiveHermes(payload.ChainID)
+	if err != nil {
+		log.Err(err).Msg("failed to get active Hermes")
+		return
+	}
+	hermesChannel, err := hcr.Fetch(payload.ChainID, payload.ID, hermes)
+	if err != nil {
+		log.Err(err).Msg("failed to get hermes channel info")
+		return
+	}
+
+	unsettledBalance := hermesChannel.UnsettledBalance()
+	if unsettledBalance.Cmp(big.NewInt(0)) == 0 {
+		data, err := hcr.hermesCaller.GetProviderData(payload.ChainID, payload.ID.Address)
+
+		if err != nil {
+			log.Err(err).Msg("failed to get provider data")
+		}
+
+		if data.LatestPromise.Amount != nil && data.LatestPromise.Amount.Cmp(big.NewInt(0)) != 0 {
+			R := generateR()
+			hashlock := ethcrypto.Keccak256(R)
+			details := rRecoveryDetails{
+				R:           hex.EncodeToString(R),
+				AgreementID: big.NewInt(0),
+			}
+
+			bytes, err := json.Marshal(details)
+			if err != nil {
+				log.Err(err).Msgf("could not marshal R recovery details: %s", err)
```

Comment:
You're already printing the error in `Err(err)`, no reason to also include it in the message. Same for other places.